src/server/pegasus_server_impl.cpp

```diff
@@ -2634,6 +2635,10 @@ void pegasus_server_impl::update_usage_scenario(const std::map<std::string, std:
                            old_usage_scenario,
                            new_usage_scenario);
         }
+    } else {
+        // When an old db is opened and the conf is changed, the options related to usage scenario
+        // need to be recalculated with new values.
+        recalculate_data_cf_options(_tmp_data_cf_opts);
```

Comment:
If I didn't miss any thing, I found this function will not be called when open an exist db, right?

Reply:
No, no matter what the last `usage scenario` is, it will change from 'normal' to the corresponding mode, so it will be executed once.
here is a cyclic task, it will get the current `usage scenario`, in order to only execute once when the db is opened, so `is_need_update_data_cf_opts` variable is added to mark whether `recalculate_data_cf_options` has been called